typing/parmatch.ml

```diff
@@ -127,8 +127,10 @@ let clean_copy ty =
 let get_type_path ty tenv =
   let ty = Ctype.repr (Ctype.expand_head tenv (clean_copy ty)) in
   match ty.desc with
-  | Tconstr (path,_,_) -> path
-  | _ -> fatal_error "Parmatch.get_type_path"
+  | Tconstr (path,_,_) -> Some path
+  | _ -> None
+   (* Same as PR#6394: we have an incoherence due to recursive module. It
+      will result in a type error later on. *)
```

Comment:
One does indeed get a fatal error here on the small example from [MPR#6394](https://caml.inria.fr/mantis/view.php?id=6394), however I feel like this change should be proposed independently of the current GPR.
The change is indeed somewhat unrelated from the rest of the PR and should IMO be discussed independently.